main/modes/mode_manual.c

```diff
@@ -0,0 +1,400 @@
+                plotLine(manual->disp, x0, y0, x1, y1, manual->fgColor, 0);
+                // Top-Left -> Top Right
+                plotLine(manual->disp, x0, y0, x1, y0, manual->fgColor, 0);
+                // Top-Right -> Bottom Left
+                plotLine(manual->disp, x1, y0, x0, y1, manual->fgColor, 0);
+                // Bottom-Left -> Bottom-Right
+                plotLine(manual->disp, x0, y1, x1, y1, manual->fgColor, 0);
+            }
+            else
+            {
+                manual->navigate = NONE;
+            }
+        }
+    }
+
+    fillDisplayArea(manual->disp, 0, 0, manual->disp->w, MANUAL_TOP_MARGIN, manual->bgColor);
+    fillDisplayArea(manual->disp, 0, 0, MANUAL_SIDE_MARGIN, manual->disp->h, manual->bgColor);
+    fillDisplayArea(manual->disp, manual->disp->w - MANUAL_SIDE_MARGIN + 1, 0, manual->disp->w, manual->disp->h, manual->bgColor);
+    fillDisplayArea(manual->disp, 0, manual->disp->h - MANUAL_BOTTOM_MARGIN + 1, manual->disp->w, manual->disp->h, manual->bgColor);
+
+    plotLine(manual->disp, 0, manual->disp->h - MANUAL_BOTTOM_MARGIN + 1, manual->disp->w, manual->disp->h - MANUAL_BOTTOM_MARGIN + 1, manual->fgColor, 0);
+
+    //  || manual->curPage->next != NULL || curPage->chapter < lastChapter
+    if (hasNext)
+    {
+        // Draw a "next page" arrow
+        drawWsg(manual->disp, &manual->arrowWsg, manual->disp->w - 40 - manual->arrowWsg.w, manual->disp->h - (MANUAL_BOTTOM_MARGIN - manual->arrowWsg.h) / 2 - manual->arrowWsg.h, false, false, 90);
+    }
+
+    if (hasPrev)
+    {
+        // Draw a "prev page" arrow
+        drawWsg(manual->disp, &manual->arrowWsg, 40, manual->disp->h - (MANUAL_BOTTOM_MARGIN - manual->arrowWsg.h) / 2 - manual->arrowWsg.h, false, false, 270);
+    }
+
+
+}
+
+void manualNextChapter(void)
+{
+    if (manual->chapter != lastChapter)
+    {
+        manual->chapter++;
+        manual->curPage = manual->curPage->next;
+        manualLoadText(false);
+    }
+}
+
+void manualPrevChapter(void)
+{
+    if (manual->chapter > chapters)
+    {
+        manual->chapter--;
+        manual->curPage = manual->curPage->prev;
+        manualLoadText(true);
+    }
+}
+
+void manualNextPage(void)
+{
+    // if there is a next page, and it is in the same chapter, go to it
+    // if there is not a next page, or it is in a different chapter, go to next chapter
+    if (manual->curPage->next != NULL && ((page_t*)manual->curPage->next->val)->chapter == manual->chapter)
+    {
+        manual->curPage = manual->curPage->next;
+    }
+    else
+    {
+        manualNextChapter();
+    }
+}
+
+void manualPrevPage(void)
+{
+    if (manual->curPage->prev != NULL && ((page_t*)manual->curPage->prev->val)->chapter == manual->chapter)
+    {
+        manual->curPage = manual->curPage->prev;
+    }
+    else
+    {
+        manualPrevChapter();
+    }
+}
+
+void manualButtonCb(buttonEvt_t* evt)
+{
+    if (evt->down)
+    {
+        switch (evt->button)
+        {
+            case UP:
```

Comment:
What's the point of chapters if you can only navigate page by page?

Reply:
That's what I was trying to add most recently, but I was doing it in a dumb way and didn't quite have time to finish after I started doing it in a not-dumb way.

Reply:
> Have you coordinated with thaeli about this?

I talked to thaeli earlier today, and I got her raw Google form submission csv, so I can make a rough pass at slapping that into markdown formatting tomorrow